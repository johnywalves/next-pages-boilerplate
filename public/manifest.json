{
  "name": "Boilerplate to React, NextJS and TypeScript",
  "short_name": "React Next - Boilerplate",
  "description": "Boilerplate with React, NextJS, Typescript, Storybook and Styled Components to get started fast and with minimal configuration",
  "background_color": "#0e4266",
  "theme_color": "#e0138c",
  "display": "fullscreen",
  "start_url": "/",
  "public_url": "",
  "icons": [
    {
      "src": "/img/icon-192.png",
      "type": "image/png",
      "sizes": "192x192"
    },
    {
      "src": "/img/icon-512.png",
      "type": "image/png",
      "sizes": "512x512"
    }
  ],
  "thumbnail": "https://raw.githubusercontent.com/johnywalves/react-next-boilerplate/main/public/img/thumbnail.png"
}
