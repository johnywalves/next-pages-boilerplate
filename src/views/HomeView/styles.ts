import styled from 'styled-components'

export const LogoHomeView = styled.img`
  width: min(var(--300px), 50%);
`

export const TitleHomeView = styled.h1`
  color: var(--color-primary);
  font-size: 2rem;
`

export const DescriptionHomeView = styled.h2`
  font-size: 1.5rem;
  line-height: 140%;
  font-weight: 400;
  width: min(var(--720px), 80%);
`

export const IllustrationHomeView = styled.img`
  margin: var(--20px) 0;
  width: min(var(--320px), 100%);
`

export const JustHomeView = styled.p`
  font-size: 1.5rem;
`

export const CodeHomeView = styled.code`
  background-color: var(--color-code);
  font-size: 1rem;
  font-weight: 900;
  padding: 1rem;
  border: 1px solid var(--color-text);
`
