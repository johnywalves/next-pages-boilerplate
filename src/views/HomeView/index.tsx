import Layout from 'components/Layout'

import manifest from '../../../public/manifest.json'
import {
  LogoHomeView,
  TitleHomeView,
  DescriptionHomeView,
  IllustrationHomeView,
  JustHomeView,
  CodeHomeView
} from './styles'

const HomeView = ({
  title = manifest.name,
  description = manifest.description
}) => (
  <Layout>
    <LogoHomeView
      src="/img/logo.svg"
      alt="Imagem de um átomo e React Avançado escrito ao lado."
    />
    <TitleHomeView>{title}</TitleHomeView>
    <DescriptionHomeView>{description}</DescriptionHomeView>
    <IllustrationHomeView src="/img/hero.svg" alt="Uma moça de jaqueta " />
    <JustHomeView>just run</JustHomeView>
    <CodeHomeView>
      <p>
        [npx or yarn] create next-app -e
        https://github.com/johnywalves/react-next-boilerplate
      </p>
    </CodeHomeView>
  </Layout>
)

export default HomeView
