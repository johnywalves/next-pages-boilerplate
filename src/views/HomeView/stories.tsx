import { Story, Meta } from '@storybook/react/types-6-0'

import HomeView from '.'
import manifest from '../../../public/manifest.json'

export default {
  title: 'HomeView',
  component: HomeView,
  args: {
    title: 'Default'
  }
} as Meta

const Template: Story = (args) => <HomeView {...args} />

export const Basic = Template.bind({})

Basic.args = {
  title: manifest.name
}
