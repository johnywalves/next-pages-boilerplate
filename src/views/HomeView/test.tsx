import { render, screen } from '@testing-library/react'

import HomeView from '.'
import manifest from '../../../public/manifest.json'

describe('<HomeView />', () => {
  it('should render the heading', () => {
    const { container } = render(<HomeView />)

    expect(
      screen.getByRole('heading', { name: manifest.name })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the colors correctly', () => {
    const { container } = render(<HomeView />)

    expect(container.firstChild).toHaveStyleRule(
      'background-color',
      'var(--color-background)'
    )
  })
})
