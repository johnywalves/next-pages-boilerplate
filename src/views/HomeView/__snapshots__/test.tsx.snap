// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeView /> should render the heading 1`] = `
.c0 {
  background-color: var(--color-background);
  color: var(--color-text);
  width: 100%;
  min-height: 100%;
  padding: var(--30px);
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  gap: var(--16px);
}

.c0 button {
  background-color: transparent;
  color: var(--color-text);
  border: 1px solid var(--color-primary);
  padding: var(--16px) var(--24px);
  font-size: 1.25rem;
  font-weight: 600;
  cursor: pointer;
  -webkit-transition: background-color 0.25s;
  transition: background-color 0.25s;
}

.c0 button:hover {
  background-color: var(--color-primary);
  color: var(--color-text);
}

.c1 {
  width: min(var(--300px),50%);
}

.c2 {
  color: var(--color-primary);
  font-size: 2rem;
}

.c3 {
  font-size: 1.5rem;
  line-height: 140%;
  font-weight: 400;
  width: min(var(--720px),80%);
}

.c4 {
  margin: var(--20px) 0;
  width: min(var(--320px),100%);
}

.c5 {
  font-size: 1.5rem;
}

.c6 {
  background-color: var(--color-code);
  font-size: 1rem;
  font-weight: 900;
  padding: 1rem;
  border: 1px solid var(--color-text);
}

<main
  class="c0"
>
  <img
    alt="Imagem de um átomo e React Avançado escrito ao lado."
    class="c1"
    src="/img/logo.svg"
  />
  <h1
    class="c2"
  >
    Boilerplate to React, NextJS and TypeScript
  </h1>
  <h2
    class="c3"
  >
    Boilerplate with React, NextJS, Typescript, Storybook and Styled Components to get started fast and with minimal configuration
  </h2>
  <img
    alt="Uma moça de jaqueta "
    class="c4"
    src="/img/hero.svg"
  />
  <p
    class="c5"
  >
    just run
  </p>
  <code
    class="c6"
  >
    <p>
      [npx or yarn] create next-app -e https://github.com/johnywalves/react-next-boilerplate
    </p>
  </code>
</main>
`;
